homeassistant/components/cover/homekit_controller.py

```diff
@@ -0,0 +1,305 @@
+        self._position = None
+        self._tilt_position = None
+        self._hold = None
+        self._obstruction_detected = None
+        self.lock_state = None
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._state is not None
+
+    def update_characteristics(self, characteristics):
+        """Synchronise the Cover state with Home Assistant."""
+        # pylint: disable=import-error
+        from homekit.model.characteristics import CharacteristicsTypes
+
+        for characteristic in characteristics:
+            ctype = characteristic['type']
+            ctype = CharacteristicsTypes.get_short(ctype)
+            if ctype == "position.state":
+                self._chars['position.state'] = \
+                    characteristic['iid']
+                if 'value' in characteristic:
+                    self._state = \
+                        CURRENT_WINDOW_STATE_MAP[characteristic['value']]
+            elif ctype == "position.current":
+                self._chars['position.current'] = \
+                    characteristic['iid']
+                self._position = characteristic['value']
+            elif ctype == "position.target":
+                self._chars['position.target'] = \
+                    characteristic['iid']
+            elif ctype == "position.hold":
+                self._chars['position.hold'] = characteristic['iid']
+                if 'value' in characteristic:
+                    self._hold = characteristic['value']
+            elif ctype == "vertical-tilt.current":
+                self._chars['vertical-tilt.current'] = characteristic['iid']
+                self._tilt_position = characteristic['value']
+            elif ctype == "horizontal-tilt.current":
+                self._chars['horizontal-tilt.current'] = characteristic['iid']
+                self._tilt_position = characteristic['value']
+            elif ctype == "vertical-tilt.target":
+                self._chars['vertical-tilt.target'] = \
+                    characteristic['iid']
+            elif ctype == "horizontal-tilt.target":
+                self._chars['vertical-tilt.target'] = \
+                    characteristic['iid']
+            elif ctype == "obstruction-detected":
+                self._chars['obstruction-detected'] = characteristic['iid']
+                self._obstruction_detected = characteristic['value']
+            elif ctype == "name":
+                self._chars['name'] = characteristic['iid']
+                if 'value' in characteristic:
+                    self._name = characteristic['value']
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        supported_features = (
+            SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION)
+
+        if self._tilt_position is not None:
```

Comment:
`self._tilt_position` will never go back to `None` if it's ever set to not `None`, right? We don't want supported features to change during entity lifetime. So is this basically updated on first update and then not changed?

Reply:
I've added a check before setting `self._tilt_position` to make sure this does not happen.